src/ontology/uberon.Makefile

```diff
@@ -934,40 +934,59 @@ subsets/life-stages-core.owl: uberon.owl
+#            all classes that are unsatisfiable because of that
+#            assertion.
+# - precise: alternative strategy; given a root class R and a taxon T,
+#            iterate through all subclasses C of R and include classes
+#            for which the expression 'C and in_taxon some T' is
+#            satisfiable; about 3-5 times slower than 'default'.
+TAXON_SUBSET_STRATEGY = default
+
+TAXON_SUBSET_ROOTS = UBERON:0001062 UBERON:0000000
+
+# Create a taxon-specific subset
+.PRECIOUS: subsets/%-view.owl
+subsets/%-view.owl: uberon.owl | all_robot_plugins
+	$(ROBOT) uberon:create-species-subset --input $< \
+		                              --taxon $(TAXON_ID_$*) \
+		                              --strategy $(TAXON_SUBSET_STRATEGY) \
+		                              --reasoner ELK \
+		                              $(foreach root,$(TAXON_SUBSET_ROOTS),--root $(root)) \
+		 reason --reasoner ELK --equivalent-classes-allowed all \
+		        --exclude-tautologies structural \
 		 relax \
 		 remove --axioms equivalent \
 		 relax \
-		 filter --term-file tmp/simple-slim-seed.txt \
-		        --select "annotations ontology anonymous self" --trim true --signature true \
-		 reduce -r ELK \
-		 query --update ../sparql/inject-subset-declaration.ru \
+		 reduce --reasoner ELK \
 		 annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) \
-		 convert -f ofn -o $@.tmp.owl && mv $@.tmp.owl $@
-.PRECIOUS: subsets/%-basic.owl
-
-$(TAXMODSDIR)/uberon-taxmod-%.obo: $(TAXMODSDIR)/uberon-taxmod-%.owl
-	$(OWLTOOLS) $< --remove-imports-declarations -o -f obo --no-check $@.tmp && grep -v ^owl $@.tmp > $@
-
-# added --allowEquivalencies, see https://github.com/geneontology/go-ontology/issues/12926
-$(TMPDIR)/uberon-taxmod-%.owl: uberon.owl
-	$(OWLTOOLS) $< --reasoner elk --make-species-subset --perform-macro-expansion false -t NCBITaxon:$* \
-		    --assert-inferred-subclass-axioms --allowEquivalencies --useIsInferred --remove-dangling \
-		    --set-ontology-id $(URIBASE)/uberon/subsets/$@ -o $@ 2>&1 > $@.log
-.PRECIOUS: $(TMPDIR)/uberon-taxmod-%.owl
+		 convert --format ofn --output $@
+
+# Create a component containing oboInOwl:inSubset annotations that mark
+# the classes that belong to a taxon-specific subset
+subsets/%-tags.ofn: uberon.owl | all_robot_plugins
+	$(ROBOT) uberon:create-species-subset --input $< \
+		                              --taxon $(TAXON_ID_$*) \
+		                              --strategy $(TAXON_SUBSET_STRATEGY) \
+		                              --reasoner ELK \
+		                              $(foreach root,$(TAXON_SUBSET_ROOTS),--root $(root)) \
+		                              --no-remove \
+		                              --prefix 'uberon: http://purl.obolibrary.org/obo/uberon/core#' \
+		                              --subset-name uberon:$*_subset \
+		                              --only-tag-in UBERON: --only-tag-in GO: \
+		                              --write-tags-to $@
```

Comment:
I personally have a very weak preference to avoid Java code however well designed because of the lack of people being able to maintain it, but this is not a review comment now, just an FYI that you should ignore. I am assuming the Java code does some things beyond the capability of a simple SPARQL query here.

Reply:
I personally have a very strong preference for a well-designed, small piece of code (in any language) that does one thing and does it well, over a complicated pipeline that chains two or three tools and/or needs several intermediate representations of the ontology in order to achieve something. :)

Reply:
Fair enough :P Lets do the dance on a case by case basis. In this case here I agree!